query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PromQLRangeQuery.java

```diff
@@ -0,0 +1,21 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Singular;
+
+@Getter
+@Builder
+class PromQLRangeQuery {
+  @NonNull @Singular private List<String> queries;
+
+  @NonNull private Instant startTime;
```

Comment:
QueryTimeRange maybe?

Reply:
Skipping any comments on the model classes in this pr, @kotharironak have addressed them on the other one #110